### DISCUSSION TITLE
Connect image and gallery field in post model to image model

### DISCUSSION THREAD

**Discussion by IRediTOTO** [prisma/prisma#11867] [Q&A] [answered]
Hi, I 2 model like this
```
model Deal {
id Int @id
image Image  // what is the right way to connect image and gallery to Image model? 
gallery Image[]
}
```
```
model Image {
id String @id
Deal[]
}
```

**Comment by nurul3101** [comment, marked as answer]
Hey @IRediTOTO :wave:,
Could you please provide more context on what kind of relationship exists between `Deal` and `Image` model? Is it one-to-one, one-to-many, many-to-many?

This is an example in which Deal model has two relationships with Image model which I think is more appropriate according to your snippet.

```prisma
model Deal {
  id      Int     @id
  image   Image[] @relation(name: "DealImages")
  gallery Image[] @relation(name: "GalleryImages")
}

model Image {
  id            Int @id
  dealImageId   Int?
  dealGalleryId Int?
  dealImages    Deal?  @relation("DealImages", fields: [dealImageId], references: [id])
  galleryImages Deal?  @relation("GalleryImages", fields: [dealGalleryId], references: [id])
}
```

Referring [Disambiguating relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#disambiguating-relations) should be useful in case of defining two relations with the same two models.

**Reply by IRediTOTO** [reply]
Thank you so much <3

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
